### PR TITLE
Copy x86_64-apple-ios-simulator swiftmodule files for AWSMobileClient…

### PR DESCRIPTION
**Pull request by wooj2**
Tested locally on my box by importing in generated build artifacts into a new project file.  Made sure to compile and run code that was calling into the AWSMobileClient.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,6 @@
 # AWS Mobile SDK for iOS CHANGELOG
+## Unreleased
+- Fixed issue 2274 where we were not packaging x86_64-apple-ios-simulator.swiftmodule files for binary releases
```
Comment (palpatim) [inline CHANGELOG.md]: nit: include issue link in changelog notes. See previous examples.

File: CircleciScripts/package_sdk.sh [hunk 2]
```diff
@@ -74,6 +74,7 @@ mkdir -p $FRAMEWORK_DIR
 echo "copy framework"
 echo "builtFramework/Release-iphoneos/${project_name}.framework $FRAMEWORK_DIR"
```
Comment (palpatim) [inline CircleciScripts/package_sdk.sh]: nit: remove, we should use `set -v` if we want to echo commands to console as we execute

Comment (wooj2) [reply CircleciScripts/package_sdk.sh]: Updated.

File: CHANGELOG.md (hunk 1 above)
Comment (wooj2) [reply CHANGELOG.md]: Updated.

**Pull request merged**